gateway-ha/src/main/java/io/trino/gateway/ha/router/RESTApiRoutingGroupSelector.java

```diff
@@ -0,0 +1,154 @@
+                .addAll(defaultBlacklistHeaders)
+                .addAll(rulesRESTConfiguration.getBlackListHeaders())
+                .build();
+
+//        this.requestAnalyzerConfig = requestAnalyzerConfig;
+//        trinoRequestUserProvider = new TrinoRequestUser.TrinoRequestUserProvider(requestAnalyzerConfig);
+
+        URI uri = null;
+        try {
+            requireNonNull(rulesRESTConfiguration.getUrlPath(),
+                    "Invalid url provided, using routing group header as default.");
+            uri = new URI(rulesRESTConfiguration.getUrlPath());
+        }
+        catch (Exception e) {
+            throw new RuntimeException("Invalid url provided, using "
+                    + "routing group header as default.", e);
+        }
+        this.uri = uri;
+        this.httpClient = new JettyHttpClient(new HttpClientConfig());
+    }
+
+    @Override
+    public String findRoutingGroup(HttpServletRequest servletRequest)
+    {
+        Request request;
+        JsonBodyGenerator<RoutingGroupRESTApiBody> requestBodyGenerator;
+        try {
+            RoutingGroupRESTApiBody requestBody = createRequestBody(servletRequest);
+            requestBodyGenerator = jsonBodyGenerator(ROUTING_GROUP_REST_API_BODY_JSON_CODEC, requestBody);
+            request = preparePost()
+                    .addHeader(CONTENT_TYPE, JSON_UTF_8.toString())
+                    .addHeaders(getValidHeaders(servletRequest))
+                    .setUri(uri)
+                    .setBodyGenerator(requestBodyGenerator)
+                    .build();
+
+            // Execute the request and get the response
+            RoutingGroupRESTApiResponse response = httpClient.execute(request, ROUTING_GROUP_REST_API_JSON_RESPONSE_HANDLER);
+
+            // Check the response and return the routing group
+            if (response == null) {
+                throw new RuntimeException("Unexpected response: null");
+            }
+            else if (response.getErrors() != null && !response.getErrors().isEmpty()) {
+                throw new RuntimeException("Response with error: " + String.join(", ", response.getErrors()));
+            }
+            return response.getRoutingGroup();
+        }
+        catch (Exception e) {
+            log.error(e, "Error occurred while retrieving routing group "
+                    + "from Routing Rules API API at " + uri);
+        }
+        return servletRequest.getHeader(ROUTING_GROUP_HEADER);
+    }
+
+    private RoutingGroupRESTApiBody createRequestBody(HttpServletRequest request)
+    {
+        // TODO: Serialize TrinoQueryProperties, TrinoRequestUser
+//        TrinoQueryProperties trinoQueryProperties = null;
+//        TrinoRequestUser trinoRequestUser = null;
+//        if (requestAnalyzerConfig.isAnalyzeRequest()) {
+//            trinoQueryProperties = new TrinoQueryProperties(request, requestAnalyzerConfig);
+//            trinoRequestUser = trinoRequestUserProvider.getInstance(request);
+//        }
```

Comment:
@willmostly PTAL. As of now, `TrinoQueryProperties, TrinoRequestUser` is not serializable using `@JsonProperty` so I had to comment out. 
Should i just remove code? or just keep it commented out until you serialize it.